File: 2473-max-sum-of-a-pair-with-equal-sum-of-digits/max-sum-of-a-pair-with-equal-sum-of-digits.cpp
class Solution {
public:
    int maximumSum(vector<int>& nums) {
        int dp[82] = {0};
        int ans = -1;

        for (int x : nums) {
            int sum = 0, temp = x;

            while (temp != 0) {
                sum += temp % 10;
                temp /= 10;
            }

            if (dp[sum] != 0) {
                ans = std::max(ans, x + dp[sum]);
            }
            dp[sum] = std::max(dp[sum], x);
        }

        return ans;
    }
};

File: 2473-max-sum-of-a-pair-with-equal-sum-of-digits/max-sum-of-a-pair-with-equal-sum-of-digits.py
class Solution:
    def maximumSum(self, nums: List[int]) -> int:
        dp = [0] * 82
        ans = -1

        for x in nums:
            dsum = sum(int(d) for d in str(x))
            if dp[dsum] != 0:
                ans = max(ans, x + dp[dsum])
            dp[dsum] = max(dp[dsum], x)

        return ans
